biorxiv/word_vector_experiment/07_biorxiv_visualize_embeddings_pretrained.py

```diff
@@ -0,0 +1,195 @@
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "doc2vec": (
+        pd.read_csv(
+            "output/embedding_output/umap/biorxiv_umap_300_doc2vec.tsv", sep="\t"
+        )
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "umap1": "last",
+                "umap2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "pretrained": (
+        pd.read_csv(
+            "output/embedding_output/umap/biorxiv_umap_300_pretrained.tsv", sep="\t"
+        )
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "umap1": "last",
+                "umap2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+}
+
+# Keep only current versions of articles
+biorxiv_pca_models_latest = {
+    "original": (
+        pd.read_csv("output/embedding_output/pca/biorxiv_pca_300.tsv", sep="\t")
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "pca1": "last",
+                "pca2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "doc2vec": (
+        pd.read_csv("output/embedding_output/pca/biorxiv_pca_300_doc2vec.tsv", sep="\t")
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "pca1": "last",
+                "pca2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+    "pretrained": (
+        pd.read_csv(
+            "output/embedding_output/pca/biorxiv_pca_300_pretrained.tsv", sep="\t"
+        )
+        .groupby("doi")
+        .agg(
+            {
+                "doi": "last",
+                "document": "last",
+                "pca1": "last",
+                "pca2": "last",
+                "category": "last",
+            }
+        )
+        .reset_index(drop=True)
+    ),
+}
+
+# # UMAP of the Documents
+
+g = (
+    p9.ggplot(biorxiv_umap_models_latest["original"])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.labs(title="UMAP of BioRxiv (Word dim: 300)", color="Article Category")
+)
+g.save("output/embedding_output/umap/figures/biorxiv_umap_300.png", dpi=500)
+print(g)
+
+g = (
+    p9.ggplot(biorxiv_umap_models_latest["doc2vec"])
+    + p9.aes(x="umap1", y="umap2", color="factor(category)")
+    + p9.geom_point()
+    + p9.labs(title="UMAP of BioRxiv (Doc2vec Word dim: 300)", color="Article Category")
+)
+g.save("output/embedding_output/umap/figures/biorxiv_umap_300_doc2vec.png", dpi=500)
+print(g)
+
+g = (
```

Comment:
I'm having a little trouble interpreting these plots, particularly what the difference between UMAP vs PCA in this case. It looks like you're able to capture more of the difference between article categories in UMAP vs PCA, though I'm not sure why this is the case. Also in general it feels like articles are pretty squished towards 0 in both visualizations - is this expected?

Reply:
>I'm having a little trouble interpreting these plots, particularly what the difference between UMAP vs PCA in this case. It looks like you're able to capture more of the difference between article categories in UMAP vs PCA, though I'm not sure why this is the case. Also in general it feels like articles are pretty squished towards 0 in both visualizations - is this expected?

The difference between UMAP and PCA are the way the low dimensional space is constructed. UMAP is non-linear and PCA is linear. The idea here is to visually show the differences between doc2vec, word2vec and pretraining. Doc2vec clusters documents a whole lot differently than word2vec, while pretraining doesn't change anything (argument against using a pretrained model for my response to reviewers). 

In terms of the 0 squishing I'd speculate that a lot of preprints contain similar concepts (e.g. sequencing or other general approaches related to that idea)  that are captured by these latent embeddings from word2vec/doc2vec. Is there any meaning in this? Probably not as this is just for visual purposes rather than in depth meaning.

Reply:
I see. I think I was trying to compare the results using UMAP vs PCA. It does look like you have the same takeaway message from either method, I was just noticing that you got a bit better separation between clusters using UMAP which is where I was coming from.

Reply:
>I see. I think I was trying to compare the results using UMAP vs PCA. It does look like you have the same takeaway message from either method, I was just noticing that you got a bit better separation between clusters using UMAP which is where I was coming from.

Yeah. The beauty of UMAP. It is better at separating clusters than PCA due to its non-linearity. I'll update my notebook description to be clear on the intent for this analysis.